src/pkg-dep.stanza

```diff
@@ -0,0 +1,171 @@
+defpackage slm/pkg-dep:
+  import core
+  import collections
+
+  import maybe-utils
+  import term-colors
+  import toml
+
+  import slm/conan-utils
+  import slm/dependency
+  import slm/errors
+  import slm/flags
+  import slm/logging
+  import slm/platforms
+  import slm/process-utils
+  import slm/string-utils
+  import slm/utils
+
+doc: \<DOC>
+Define a Pkg Dependency representing a pre-compiled stanza pkg archive
+stored in a conan/artifactory repository
+
+This dependency is intended to define a pre-compiled archive
+of stanza pkg binaries which can be linked into an slm project.
+This archive may be stored on a remote conan/artifactory server.
+<DOC>
+public defstruct PkgDependency <: Dependency & Hashable & Equalable:
+  name:String with: (
+    as-method => true
+    ensure => ensure-not-empty!
+    )
+  version: String
+  type: String
+  options: TomlTable
+
+defmethod recursive (d:PkgDependency) -> True|False :
+  ; a pkg can have recursive deps
+  true
+
+defmethod path (d:PkgDependency) -> String:
+  path-join(SLM_DEPS_DIR, name(d))
+
+defmethod equal? (a:PkgDependency, b:PkgDependency) -> True|False :
+  name(a) == name(b) and version(a) == version(b) and type(a) == type(b) and to-string(options(a)) == to-string(options(b))
+
+defmethod hash (x:PkgDependency) -> Int :
+  defn toml-hash (t:TomlTable|TomlValue) -> Int:
+    match(t):
+      (i:Int): hash(i)
+      (s:String): hash(s)
+      (tt:TomlTable):
+        sum $ for kvp in tt seq:
+          toml-hash(key(kvp)) + toml-hash(value(kvp))
+  hash(name(x)) + hash(version(x)) + hash(type(x)) + toml-hash(options(x))
```

Comment:
Does `hashable => true` on the defstruct definition not work for `PkgDependency` ? 

```
public defstruct PkgDependency <: Dependency & Hashable & Equalable:
  name:String with: (
    as-method => true
    ensure => ensure-not-empty!
    )
  version: String
  type: String
  options: TomlTable
with:
  hashable => true
```

Reply:
Maybe so.  I was copying code from another file.  I'll try hashable.